introduce buffer specific b:neomake_FT_enabled_makers

davidosomething opened the pull request:
Example use case:

- Project has blog posts in markdown format that need to be linted using `prose` linter
- Project has a `readme.md` that should not be `prose` linted
- Set `b:neomake_markdown_enabled_makers = [ 'prose', 'markdownlint' ]` on the blog post buffer, set `b:neomake_markdwon_enabled_makers = [ 'markdownlint' ]` on the readme.md buffer

blueyed commented on autoload/neomake.vim:
```diff
@@ -288,19 +290,30 @@ function! neomake#GetEnabledMakers(...) abort
-        if exists(varname)
-            let enabled_makers = eval(varname)
-        else
+        unlet! l:enabled_makers
```
Why is the `unlet` necessary here?

blueyed replied:
Ah, the `!exists('l:enabled_makers')` block below will set it always, and then it would be set for the next ft.

blueyed commented:
LGTM, please also add it to the documentation.

davidosomething commented:
Updated docs https://github.com/neomake/neomake/pull/483/commits/2c9aa432251715b7b3955a1b851c57893675cf52


Of note, I removed some code where I introduced `b:neomake_enabled_makers` it has an unintended side effect -- details in the modified help for `g:neomake_enabled_makers`.
Eventually I'd like to reintroduce `b:neomake_enabled_makers` since it would provide functionality like syntastic's `b:syntastic_checkers`. In another PR though.

davidosomething commented:
@blueyed what do you think, ready to merge?

blueyed commented on doc/neomake.txt:
```diff
@@ -199,12 +199,14 @@ Configure a property for a prexisting maker where property is one of 'exe',
 This setting will tell neomake which makers to use by default for the given
 filetype. Filetypes that already have makers should have a default list
-of makers already. Example: >
+of makers already. Can also be set by buffer. Example: >
```
s/Can also be set by buffer/This can also be set per buffer/ ?

blueyed commented:
Let's follow up in https://github.com/neomake/neomake/pull/503, where I've added a fixup commit.
